lib/ansible/modules/crypto/entrust/ecs_domain.py

```diff
@@ -0,0 +1,409 @@
+            return False
+
+    def request_domain(self, module):
+        if not self.check(module):
+            body = {}
+
+            body['verificationMethod'] = module.params['verification_method']
+            if module.params['verification_method'] == 'EMAIL':
+                emailMethod = {}
+                if module.params['verification_email']:
+                    emailMethod['emailSource'] = 'SPECIFIED'
+                    emailMethod['email'] = module.params['verification_email']
+                else:
+                    emailMethod['emailSource'] = 'INCLUDE_WHOIS'
+                body['emailMethod'] = emailMethod
+            # Only populate domain name in body if it is not an existing domain
+            if not self.domain_status:
+                body['domainName'] = module.params['domain_name']
+            try:
+                if not self.domain_status:
+                    self.ecs_client.AddDomain(clientId=module.params['client_id'], Body=body)
+                else:
+                    self.ecs_client.ReverifyDomain(clientId=module.params['client_id'], domain=module.params['domain_name'], Body=body)
+
+                time.sleep(5)
+                result = self.ecs_client.GetDomain(clientId=module.params['client_id'], domain=module.params['domain_name'])
+
+                # It takes a bit of time before the random values are available
+                if module.params['verification_method'] == 'DNS' or module.params['verification_method'] == 'WEB_SERVER':
+                    for i in range(4):
+                        # Check both that random values are now available, and that they're different than were populated by previous 'check'
+                        if module.params['verification_method'] == 'DNS':
+                            if result.get('dnsMethod') and result['dnsMethod']['recordValue'] != self.dns_contents:
+                                break
+                        elif module.params['verification_method'] == 'WEB_SERVER':
+                            if result.get('webServerMethod') and result['webServerMethod']['fileContents'] != self.file_contents:
+                                break
+                    time.sleep(10)
+                    result = self.ecs_client.GetDomain(clientId=module.params['client_id'], domain=module.params['domain_name'])
+                self.changed = True
+                self.set_domain_details(result)
+            except RestOperationException as e:
+                module.fail_json(msg='Failed to request domain validation from Entrust (ECS) {0}'.format(e.message))
+
+    def dump(self):
+        result = {
+            'changed': self.changed,
+            'client_id': self.client_id,
+            'domain_status': self.domain_status,
+        }
+
+        if self.verification_method:
+            result['verification_method'] = self.verification_method
+        if self.ov_eligible:
+            result['ov_eligible'] = self.ov_eligible
```

Comment:
Above, you wrote that `ov_eligible` is returned when `success and I(domain_status) is C(APPROVED), C(RE_VERIFICATION) or C(EXPIRING).`. But here, you only return it if it has value `True`.